admin-app/src/pages/ViewBranchQueues/ViewBranchQueues.jsx

```diff
@@ -0,0 +1,152 @@
+                if (response.success) {
+                    setBranches(response.data);
+                } else {
+                    console.error('Error fetching branches:', response.error);
+                }
+            } catch (error) {
+                console.error('Error fetching branches:', error);
+            }
+        }
+
+        fetchBranches();
+    }, [location, tenantCode]);
+
+    function setUpDropdowns(branch) {
+        setSelectedBranch(branch);
+        setServiceName('Select Service');
+        setQueue(originalQueue);
+    }
+
+    function setDropdownService(service) {
+        setSelectedService(service);
+        setServiceName(service.name);
+        updateQueue(service.id);
+    }
+
+    function updateQueue(serviceId) {
+        if (serviceId) {
+            const filteredQueue = originalQueue.map(queueItem => ({
+                ...queueItem,
+                tickets: Array.isArray(queueItem.tickets) ? queueItem.tickets.filter(ticket => ticket.service.id === serviceId) : []
+            }));
+            setQueue(filteredQueue);
+        } else {
+            setQueue(originalQueue);
+        }
+    }
+
+    return (
+        <div className="text-center mt-5">
+            <h2>Queue</h2>
+            <DropdownButton title={selectedBranch ? selectedBranch.name : 'Select Branch'} variant="btn btn-outline-secondary">
+                {branches.map((branch, index) => (
+                    <Dropdown.Item key={index} onClick={() => setUpDropdowns(branch)}>{branch.name}</Dropdown.Item>
+                ))}
+            </DropdownButton>
+
+            <DropdownButton style={{ marginTop: '20px' }} title={serviceName} variant="btn btn-outline-secondary">
+                {services.map((service, index) => (
+                    <Dropdown.Item key={index} onClick={() => setDropdownService(service)}>{service.name}</Dropdown.Item>
+                ))}
+            </DropdownButton>
+
+            <div style={{ marginTop: '20px' }}>
+                <Table striped bordered hover>
+                    <thead>
+                    <tr>
+                        <th>Ticket Id</th>
+                        <th>Ticket Number</th>
+                        <th>Service</th>
+                        <th>Created At</th>
+                        <th>Elapsed Time</th>
+                    </tr>
+                    </thead>
+                    <tbody>
+                    {queue.map((queueItem, index) => (
+                        queueItem.tickets.map((ticket, ticketIndex) => {
+                            const createdAt = new Date(ticket.createdAt);
+                            const currentTime = new Date();
+                            const elapsedTime = Math.floor((currentTime - createdAt) / (1000 * 60));
+
+                            return (
+                                <tr key={`${index}-${ticketIndex}`}>
+                                    <td>{ticket.id}</td>
```

Comment:
Do we really need to show the ticket id? It is unnecessary for identifying the ticket (the number is enough), and users could be confused as to what the ID does alongside the number. I would remove this column.